R/teal_data_module.R

```diff
@@ -0,0 +1,52 @@
+#' Data module for `teal` applications
+#'
+#' Creates `teal_data_module` object - a `shiny` module to supply or modify data in a `teal` application.
+#'
+#' This function creates a `shiny` module that allows for running data pre-processing code after the app starts.
+#' The body of the server function will be run in the app rather than in the global environment.
+#' This means it will be run every time the app starts, so use sparingly.
+#'
+#' Pass this module instead of a `teal_data` object in a call to [init()].
+#'
+#' See vignette \code{vignette("data-as-shiny-module", package = "teal")} for more details.
```

Comment:
How about linking to `https://insightsengineering.github.io/teal/latest-tag/articles/`? 
It seems to me vignettes are not built during a default installation. I installed with `install.packages` and I get 
```
> library(teal)
> browseVignettes("teal")
No vignettes found by browseVignettes("teal")
> vignette("data-as-shiny-module", package = "teal")
Warning message:
vignette ‘data-as-shiny-module’ not found 
```

Reply:
vignette is fine, you need to build the vignette when installing from source. Normally on CRAN or other repos they are build.

Reply:
Oh, that's why it happens. Very well :ok_hand: